drivers/pcie/host/pcie.c

```diff
@@ -548,14 +548,4 @@ static int pcie_init(void)
 }
 
 
-/*
- * If a pcie controller is employed, pcie_scan() depends on it for working.
- * Thus, pcie must be bumped to the next level
- */
-#ifdef CONFIG_PCIE_CONTROLLER
-#define PCIE_SYS_INIT_LEVEL	PRE_KERNEL_2
-#else
-#define PCIE_SYS_INIT_LEVEL	PRE_KERNEL_1
-#endif
-
-SYS_INIT(pcie_init, PCIE_SYS_INIT_LEVEL, CONFIG_PCIE_INIT_PRIORITY);
+SYS_INIT(pcie_init, EARLY, CONFIG_PCIE_INIT_PRIORITY);
```

Comment:
Seems like qemu_cortex_a53 does not like this change.

Reply:
yeah, more dependency hell

Reply:
and it seems the proposed solution here https://github.com/zephyrproject-rtos/zephyr/issues/64022 will not work, i.e, if we move pcie back to pre_kernel:

```
PRE_KERNEL_1
  __init_statics_init_pre: statics_init(NULL)
  __init___device_dts_ord_5: arm_gic_init(__device_dts_ord_5)
  __init_init_mem_domain_module: init_mem_domain_module(NULL)
  __init_app_shmem_bss_zero_pre: app_shmem_bss_zero(NULL)
  __init___device_dts_ord_17: pcie_ecam_init(__device_dts_ord_17)
  __init_pcie_init: pcie_init(NULL)
  __init___device_dts_ord_15: pl011_init(__device_dts_ord_15)
  __init_uart_console_init: uart_console_init(NULL)
```

looks like pcie_init depends on pcie_ecam_init, and uart in the case os ns16xx might depend on pcie....

Reply:
> and it seems the proposed solution here #64022 will not work, i.e, if we move pcie back to pre_kernel:

Yes, it's the neverending issue of the non-direct relationship with  DTS/Kconfig/Zephyr init entries.
I mean, as long as we cannot have all in one place (dts nodes + "software" nodes + Kconfig tweaks) to generate a proper init dependency tree, we are doomed to mangle the level/prio manually. Tough one